specifyweb/frontend/js_src/lib/components/Header/Notifications.tsx

```diff
@@ -42,16 +43,34 @@ export function Notifications({
 
   React.useEffect(() => {
     let pullInterval = INITIAL_INTERVAL;
+    let lastFetchedTimestamp: Date | undefined;
+
     const handler = (): void => {
-      if (timeout !== undefined) globalThis.clearTimeout(timeout);
+      if (timeout !== undefined) {
+        globalThis.clearTimeout(timeout);
+      }
       pullInterval = INITIAL_INTERVAL;
-      if (document.visibilityState === 'visible') doFetch();
+      if (document.visibilityState === 'visible') {
+        doFetch();
+      }
     };
+
     document.addEventListener('visibilitychange', handler);
 
     let timeout: NodeJS.Timeout | undefined = undefined;
 
-    function doFetch(): void {
+    function doFetch(since?: Date): void {
+      const startFetchTimestamp = new Date();
+      const queryString = since
+        ? encodeURIComponent(since.toISOString())
+        : // : '';
+          encodeURIComponent(new Date().toISOString());
+      //should look like this: `/notifications/messages/?since=2022-12-09%2020:35:23.000`
+      const url = formatUrl(`/notifications/messages/`, {
```

Comment:
@maxxxxxdlp , the date format passed to the back-end url is not correct, what do you advise me to use?

Reply:
does back-end accept `new Date().toJSON()`?


also, see at what the main `/api/specify/` endpoint does to the date it receives from the front-end: https://github.com/specify/specify7/blob/207df9a091c59cd04e4e4b1c3eb111c75c74f68a/specifyweb/specify/api.py#L670-L673

thus, does this work: `new Date().toISOString().replace('T', ' ')`?

also, see where back-end uses the date: https://github.com/specify/specify7/blob/207df9a091c59cd04e4e4b1c3eb111c75c74f68a/specifyweb/notifications/views.py#L20-L21. if you need to, you can modify that code

also, you can construct the date from pieces on the front-end. i.e `${date.getFullYear()}-${date.getMonth()+1}-...`

---

I also looked at the history to see what was used in there in the past (see line 34) - https://github.com/specify/specify7/commit/c34aa8e84502b325e3c9333781ba58f6fc47e694#diff-da092c6e0e273c53b4c8a86465d61826b5752a74dd6aa7cf03848be7d7aeb32f - it was just using the date from the last notifications - simple, smart.

in the process, beware of the time zones. Until https://github.com/specify/specify7/issues/641 is fixed, it's possible that there is no clean way to pass a date from front-end to back-end. If you do see the timezone causing a problem, try using the above solution of just sending back to the back-end the date of the most recent notification

Reply:
also, feel free to involve @melton-jason in the process as he did his share of fighting with django and dates 😊 so is a seasoned date samurai now 🥷.

Reply:
I'm unfortunately quite the time traveler now :hourglass_flowing_sand: :runner: